test/scroll_list/ScrollListSpec.js

```diff
@@ -1484,5 +1485,77 @@ define(function(require) {
                 });
             });
         });
+    
+        describe('_shouldPropagateBoundaryEvent', function() {
+            function testInMode(mode) {
+                it('should propagate the event when it is on left-boundary',function() {
+                    testScrollList({ mode: mode }, function(scrollList) {
+                        spyOn(scrollList.onScrollPastBoundary,'dispatch');
+                        var callback = scrollList.onScrollPastBoundary.dispatch;
+                        scrollList._shouldPropagateBoundaryEvent(null,{boundary: BoundaryTypes.LEFT});
+                        expect(callback).toHaveBeenCalled();
```

Comment:
I think it would help these tests a bunch to have the last three lines just be a function where you pass in the boundary type you are testing.